### PR TITLE
Merge 2.x for 2.4.0 release

### PR DESCRIPTION
**Pull request by dwieeb**

### CODE REVIEW

**Comment by dwieeb** [comment]
@sandstrom Here's the full PR.

**Comment by dwieeb** [comment]
@iphilgood @kheftel Would you mind glancing over this quick for your respective changes? We're pulling them into the main branch with this PR.

File: ios/Capacitor/Capacitor/Plugins/Camera.swift [hunk 1]
```diff
@@ -232,7 +234,7 @@ public class CAPCameraPlugin: CAPPlugin, UIImagePickerControllerDelegate, UINavi
         }
 
         if settings.shouldResize {
-            guard let convertedImage = resizeImage(image!) else {
+            guard let convertedImage = resizeImage(image!, false) else {
```
Comment (iphilgood) [inline ios/Capacitor/Capacitor/Plugins/Camera.swift:237]: @dwieeb this should consider the flag from the setting (`settings.preserveAspectRatio`), right?

Comment (jcesarmobile) [reply ios/Capacitor/Capacitor/Plugins/Camera.swift:237]: yeah, good catch, should be
```suggestion
            guard let convertedImage = resizeImage(image!, settings.preserveAspectRatio) else {
```

Comment (dwieeb) [reply ios/Capacitor/Capacitor/Plugins/Camera.swift:237]: Thanks, these were not easy conflicts to resolve. 😂

**Pull request merged**